src/huggingface_hub/hf_api.py

```diff
@@ -1068,17 +1083,11 @@ def create_repo(
 
         path = f"{self.endpoint}/api/repos/create"
         if token is None:
-            token = self._validate_or_retrieve_token()
-        elif not self._is_valid_token(token):
-            if self._is_valid_token(name):
-                warnings.warn(
-                    "`create_repo` now takes `token` as an optional positional argument. "
-                    "Be sure to adapt your code!",
-                    FutureWarning,
-                )
-                token, name = name, token
-            else:
-                raise ValueError("Invalid token passed!")
+            token = self._validate_or_retrieve_token(function_name="create_repo")
+        elif isinstance(token, str):
+            token, name = self._validate_or_retrieve_token(
+                token, name, function_name="create_repo"
+            )
```

Comment:
I think this can be simplified even further by just doing this:
```
token, name = self._validate_or_retrieve_token(
                token, name, function_name="create_repo"
            )
```
`_validate_or_retrieve_token` already handles case in which token is None or string, so no need to do the check here. Same applies for other functions

Reply:
I think we can move `_validate_or_retrieve_token` outside this class, move it to a utils module and use it also in the other file.

Reply:
@adrinjalali I recently pushed a new change but didn't do this yet, there's a repetition in `user.py` if we don't do this, I agree. Will push that soon.

Reply:
@adrinjalali it doesn't make any sense to put `_validate_or_retrieve_token` under endpoint helpers, the function depends heavily on other classes and functions in `hf_api.py`, so instead I called it in `user.py` like `hf_api._validate_or_retrieve_token`.